### PR TITLE
bom: 0.4.1 --> 0.5.1

### PR DESCRIPTION
**Pull request by developer-guy**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 224534` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bom</li>
  </ul>
</details>

**Review by ocfox — APPROVED** [review]

File: pkgs/tools/admin/bom/default.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 buildGoModule rec {
   pname = "bom";
-  version = "0.4.1";
+  version = "0.5.1";
```
Comment (wegank) [inline pkgs/tools/admin/bom/default.nix:9]: No hash change for an update? That's strange.

Comment (ocfox) [reply pkgs/tools/admin/bom/default.nix:9]: I wondered too, but the bom did the update.

Comment (wegank) [reply pkgs/tools/admin/bom/default.nix:9]: While this does not seem to be true upstream: https://github.com/kubernetes-sigs/bom/compare/v0.4.1...v0.5.1

Comment (wegank) [reply pkgs/tools/admin/bom/default.nix:9]: I think the new hash should be `sha256-/a+0kQn2m5JFxhlgs1pUeyWwwqVFrRbYCyUfcLsNGz8=`.

Comment (developer-guy) [reply pkgs/tools/admin/bom/default.nix:9]: I used to use `nix hash path` to calculate the sha256, what is your method to do that?

Comment (wegank) [reply pkgs/tools/admin/bom/default.nix:9]: I simply remove the old hash and `nix-build`.

**Comment by natsukium** [comment]
It would be better if the commit message is "bom: 0.4.1 -> 0.5.1".

**Review by saschagrunert — APPROVED** [review]
Changes LGTM

**Pull request merged**